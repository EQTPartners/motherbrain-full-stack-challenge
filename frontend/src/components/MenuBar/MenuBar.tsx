import React from 'react';
import { AppBar, IconButton, Toolbar, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { ArrowBack } from '@material-ui/icons';

interface MenuBarProps {
  isBackArrowVisible: boolean;
  handleBackButtonClick: () => void;
  title: string;
}

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    position: 'absolute',
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

export const MenuBar: React.FC<MenuBarProps> = (props: MenuBarProps) => {
  const classes = useStyles();

  return (
    <AppBar position="static" className={classes.root}>
      <Toolbar>
        {props.isBackArrowVisible ? (
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
            onClick={() => props.handleBackButtonClick()}
          >
            <ArrowBack />
          </IconButton>
        ) : null}
        <Typography variant="h6" className={classes.title}>
          {props.title}
        </Typography>
      </Toolbar>
    </AppBar>
  );
};
