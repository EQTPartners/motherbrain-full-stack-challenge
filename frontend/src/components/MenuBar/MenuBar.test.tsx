import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { MenuBar } from './MenuBar';

describe('MenuBar', () => {
  it('should render', () => {
    const { container } = render(
      <MenuBar isBackArrowVisible={true} handleBackButtonClick={() => {}} title="test title" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have the correct title', () => {
    const { getByLabelText } = render(
      <MenuBar isBackArrowVisible={true} handleBackButtonClick={() => {}} title="test title" />,
    );
    const element = getByLabelText('title');
    expect(element.innerHTML).toEqual('test title');
  });

  it('should not have an arrow when prop is set to false', () => {
    const { queryByLabelText } = render(
      <MenuBar isBackArrowVisible={false} handleBackButtonClick={() => {}} title="test title" />,
    );
    expect(queryByLabelText('back-icon')).not.toBeInTheDocument();
  });

  it('should have an arrow when prop is set to true', () => {
    const { getByLabelText } = render(
      <MenuBar isBackArrowVisible={true} handleBackButtonClick={() => {}} title="test title" />,
    );
    expect(getByLabelText('back-icon')).toBeInTheDocument();
  });

  it('should fire a click event when clicked', () => {
    const mockClickHandler = jest.fn();
    const { getByLabelText } = render(
      <MenuBar
        isBackArrowVisible={true}
        handleBackButtonClick={mockClickHandler}
        title="test title"
      />,
    );
    const button = getByLabelText('back-icon');
    fireEvent.click(button);
    expect(mockClickHandler).toHaveBeenCalledTimes(1);
  });
});
