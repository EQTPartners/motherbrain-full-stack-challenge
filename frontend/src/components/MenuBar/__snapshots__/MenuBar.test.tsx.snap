// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar should render 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-1 MuiPaper-elevation4"
>
  <div
    class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      aria-label="back-icon"
      class="MuiButtonBase-root MuiIconButton-root makeStyles-backIconButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <h6
      aria-label="title"
      class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
    >
      test title
    </h6>
  </div>
</header>
`;
